### PR TITLE
PN-12037

### PR DESCRIPTION
**Pull request by azhansultan**
*Added new feature file 
*Added new methods and steps in part dashboard mittente

### CODE REVIEW

File: src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+
+    public void clickUltimi6Mesi() {
+        getWebDriverWait(10).withMessage("Il bottone ultimi 6 mesi non visibile").until(ExpectedConditions.visibilityOf(ultimi6MesiBottone.get(0)));
+        logger.info("Si clicca bottone ultimi 6 mesi");
+        ultimi6MesiBottone.get(0).click();
+    }
+
+    public void clickUltimi6MesiNotificheDigitali() {
+        getWebDriverWait(10).withMessage("Il bottone ultimi 6 mesi non visibile").until(ExpectedConditions.visibilityOf(ultimi6MesiBottone.get(1)));
+        logger.info("Si clicca bottone ultimi 6 mesi");
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java:116]: what changes with the one on the line 108

File: src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+        Actions action = new Actions(driver);
+        action.keyDown(Keys.CONTROL).sendKeys("a").keyUp(Keys.CONTROL).sendKeys(Keys.BACK_SPACE).perform();
+        LocalDate dataDaInserire = LocalDate.now().minusDays(90);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+        String formattedDate = dataDaInserire.format(formatter);
+        WebTool.waitTime(1);
+        dateDa.get(0).sendKeys(formattedDate);
+    }
+
+    public void insertDataCorrettaNotificheDigitali(){
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java:182]: as before

File: src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+       getWebDriverWait(10).withMessage("Il filtro di data non visibile").until(ExpectedConditions.visibilityOf(dateDa.get(0)));
+       logger.info("Si inserisce una data errata");
+       dateDa.get(0).click();
+       Actions action = new Actions(driver);
+       action.keyDown(Keys.CONTROL).sendKeys("a").keyUp(Keys.CONTROL).sendKeys(Keys.BACK_SPACE).perform();
+       WebTool.waitTime(1);
+       dateDa.get(0).sendKeys("01/01/1111");
+    }
+
+    public void insertDataErrataNotificheDigitali(){
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java:160]: as before

File: src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+        action.keyDown(Keys.CONTROL).sendKeys("a").keyUp(Keys.CONTROL).sendKeys(Keys.BACK_SPACE).perform();
+        LocalDate dataDaInserire = LocalDate.now().minusDays(90);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+        String formattedDate = dataDaInserire.format(formatter);
+        WebTool.waitTime(1);
+        dateDa.get(1).sendKeys(formattedDate);
+    }
+
+    public void clickFiltraButton(){
+        getWebDriverWait(10).withMessage("Il bottone Filtra non cliccabile").until(ExpectedConditions.elementToBeClickable(bottoneFiltra.get(0)));
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java:196]: try to take only you will use

File: src/test/resources/feature/2-mittente/dashboard_mittenti.feature [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    And Nella pagina Statistiche Notifiche digitali cliccare sul bottone ultimi 6 mesi
+    Then Nella pagina Statistiche si visualizza il grafico Errori tecnici per tipologia
+    And Logout da portale mittente
+
+
+
+  @ApplicazioneFiltroGraficiElevataDati
+  Scenario: PN-12047 - Dashboard mittenti - Applicazione filtro grafici con PA che restituisce una quantità elevata di dati
+    Given PA - Si effettua la login tramite token exchange, e viene visualizzata la dashboard
+    When Nella pagina Piattaforma Notifiche cliccare sul bottone Statistiche
```
Comment (leleOFA) [inline src/test/resources/feature/2-mittente/dashboard_mittenti.feature:130]: work on aws?

File: src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java (hunk 1 above)
Comment (azhansultan) [reply src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java:116]: there are two different sections with filter "ultimi 6 mesi". 1 for all notifications, and 1 for digital notifications. they have different locators. other methods as well.

File: src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java (hunk 2 above)
Comment (azhansultan) [reply src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java:182]: the same

File: src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java (hunk 3 above)
Comment (azhansultan) [reply src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java:160]: the same

File: src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java (hunk 4 above)
Comment (azhansultan) [reply src/main/java/it/pn/frontend/e2e/pages/mittente/DashboardPage.java:196]: there are 2 filtra buttons. I use both

File: src/test/resources/feature/2-mittente/dashboard_mittenti.feature (hunk 5 above)
Comment (azhansultan) [reply src/test/resources/feature/2-mittente/dashboard_mittenti.feature:130]: not yet. aws fails to start

**Review by leleOFA — APPROVED** [review]

**Pull request merged**